src/php/Analyzer.php

```diff
@@ -571,6 +575,7 @@ protected function analyzeRecur(Tuple $x, NodeEnvironment $env): RecurNode {
         $frame = $env->getCurrentRecurFrame();
 
         if (!$frame) {
+            Assert::isInstanceOf($x[0], ISourceLocation::class);
```

Comment:
We can be more strict here `$x[0]` must be the Symbol 'recur'. Is is implicitly handled by the `analyze` function that has checked that and then called the this method.